sdk/core/azure-core-amqp/src/network/socket_listener.cpp

```diff
@@ -76,6 +76,13 @@ namespace Azure { namespace Core { namespace Amqp { namespace Network { namespac
     {
       m_started = false;
     }
+    if (socketlistener_stop(m_socket))
+    {
+      auto err = errno;
+      throw std::runtime_error(
+          "Could not start listener. errno=" + std::to_string(err) + ", \"" + strerror(err)
+          + "\".");
+    }
```

Comment:
Should this happen before setting `m_started` to false?

Reply:
Yes, and there's another check that is needed.